client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -12,7 +12,60 @@ import { translate } from 'i18n-calypso';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { errorNotice } from 'state/notices/actions';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+import {
+	JETPACK_ONBOARDING_SETTINGS_REQUEST,
+	JETPACK_ONBOARDING_SETTINGS_SAVE,
+} from 'state/action-types';
+import { receiveJetpackOnboardingSettings as receiveJetpackOnboardingSettingsAction } from 'state/jetpack-onboarding/actions';
+
+export const announceRequestFailure = ( { dispatch } ) =>
+	dispatch(
+		errorNotice( translate( 'Could not fetch settings from site. Please try again later.' ) )
+	);
+
+const receiveJetpackOnboardingSettings = ( { dispatch }, action, { data: settings } ) => {
+	const { siteId } = action;
+
+	if ( ! settings ) {
+		return announceRequestFailure( { dispatch }, action );
+	}
+
+	dispatch( receiveJetpackOnboardingSettingsAction( siteId, settings ) );
+};
+
+/**
+ * Dispatches a request to fetch settings for a given site
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const requestJetpackOnboardingSettings = ( { dispatch, getState }, action ) => {
+	const { siteId } = action;
+	const state = getState();
+	const token = get( state.jetpackOnboarding.credentials, [ siteId, 'token' ], null );
+	const jpUser = get( state.jetpackOnboarding.credentials, [ siteId, 'userEmail' ], null );
+
+	return dispatch(
+		http(
+			{
+				apiVersion: '1.1',
+				method: 'GET',
+				path: '/jetpack-blogs/' + siteId + '/rest-api/',
+				query: {
+					path: '/jetpack/v4/settings/',
+					query: JSON.stringify( {
+						onboarding: {
+							token,
+							jpUser,
+						},
+					} ),
```

Comment:
Doesn't `wpcom.js` already handle the serialization of the `query` parameter? I don't think we need to stringily it here or it might not even transfer properly into the URL… `?{"onboarding":{"token":…}}` instead of `?onboarding={…}`?

Reply:
> `?{"onboarding":{"token":…}}` instead of `?onboarding={…}`?

That's indeed what this is doing. Again, admittedly a case of copypasta -- this is what I've seen used pervasively with the 'proxy' endpoint, i.e. the JP endpoint to which we pass core REST API queries. I don't think we can drop it however because of that nested nature of this sort of requests -- the `query` parameter here is already something that is passed as part of a query one level higher:

`https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/SITEID/rest-api/?http_envelope=1&path=%2Fjetpack%2Fv4%2Fsettings%2F&query=%7B%22onboarding%22%3A%7B%22token%22%3A%22TOKEN6%22%2C%22jpUser%22%3A%22EMAIL%22%7D%7D&json=true&locale=de`

Reply:
(I'm not making any statement about the elegance of this, ideally there should probably be some helper that wraps the 'inner' query and passes it to the 'proxy' endpoint.)